x-pack/legacy/plugins/lens/public/app_plugin/app.tsx

```diff
@@ -48,6 +48,24 @@ function isLocalStateDirty(
   );
 }
 
+// Exported for testing purposes, sets the Kibana breadcrumbs.
+export function setBreadcrumbs(
+  chrome: {
```

Comment:
chrome should be well typed by `{Chrome} from 'ui/chrome'`

Reply:
Yeah, but that pulls in a lot of baggage that we aren't using here, and requires a comprehensive mock of Chrome. This is kind of doing the I in [SOLID](https://en.wikipedia.org/wiki/SOLID) in a way-- preferring narrow, specific interfaces to large, general ones.

Reply:
Turns out it still requires a lot more mocking, at which point, so I think it's worth defining a narrower interface in the function.

Reply:
Last comment... I was only exporting the function for testing purposes in order to avoid the complexity of testing React effects. But Wylie asked me to test the effect, so I just got rid of the function and the point about Chrome is now moot!